source/DataAccess/DADGenerator.ts

```diff
@@ -0,0 +1,58 @@
+'use strict';
+
+import IntegrationConfigFactory from '../IntegrationConfigFactory';
+import { IIntegrationConfig, IntegrationType } from '../IIntegrationConfig';
+import { writeFile } from 'fs';
+
+export default class DataAccessDoc {
+
+    private _integrationType;
+    private _queries;
+    private _header: string[] = [];
+    private _footer: string[] = [];
+
+    constructor(integrationType: IntegrationType) {
+        this._integrationType = integrationType.toString();
+        const icf = new IntegrationConfigFactory();
+        const cfg: IIntegrationConfig = icf.create(this._integrationType);
+        this._queries = cfg.queries;
+        this._header.push(`# Data Access Requirements: ${this._integrationType}`);
+        this._header.push(`DIA requires access to the following tables/fields in the ${this._integrationType} integration type.`);
+    }
+
+    public async create() {
+        const path = './docs/DataAccess/' + this._integrationType + '.md';
+        const newLine = '\n';
+        const data = this._header.join(newLine) + newLine.repeat(2) + this._createTable().join(newLine) + newLine.repeat(2) + this._footer.join(newLine);
+        await writeFile(path, data, (err) => {
+            if (err) {
+                console.log(err);
+            }
+        });
+        return;
+    }
+
+    private _getQueryInfo(): any {
+        return this._queries.map((query) => {
+            const tableName = query.name;
+
+            // First regex turns 'X.*' instances into '*', second regex removes all commas
+            const splitQuery = query.query.replace(/[^ ]*\.\*/gi, '*').replace(/,/g , '').toUpperCase().split(' ');
+
+            const fields = splitQuery.slice(1, splitQuery.indexOf('FROM'));
+            return { tableName, fields };
+        });
+    }
+
+    private _createTable(): string[] {
```

Comment:
Perhaps a term like `_createMarkdownTable` might make it a little more clear what kind of table is being generated? Many other places in our code, `createTable` often is used to refer to a persistence store in Dynamo, etc.